Cargo.toml

```diff
@@ -48,6 +48,7 @@ itertools = ">=0.10.0"
 log = "0.4"
 moka = { version = "0.9.3", default_features = false, features = ["future", "atomic64", "quanta"] }
 object_store = "0.5.0"
+percent-encoding = "2.2.0"
```

Comment:
Nice, it also looks like we're already importing this as a transitive dependency (so it's not going to result in us pulling in extra code)

```bash
~/seafowl $ cargo tree | grep percent
│   │   │   │   ├── percent-encoding v2.2.0
│   │   │   │   │   ├── percent-encoding v2.2.0
│   │   │   │   │   │   │   └── percent-encoding v2.2.0
│   │   │   │   │   │   ├── percent-encoding v2.2.0
│   │   ├── percent-encoding v2.2.0
│           └── percent-encoding v2.2.0
│   ├── percent-encoding v2.2.0
    │   │   ├── percent-encoding v2.2.0
```